### PR TITLE
python-packaging: fix provides

### PR DESCRIPTION
**Pull request by anphel31**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Rebuild python-packaging in the toolchain to resolve a circular dependency and fix the auto-generated provides for this package.

Before:
```
" Traceback (most recent call last):
"   File "/usr/lib/rpm/pythondistdeps.py", line 22, in <module>
"     from packaging.requirements import Requirement as Requirement_
" ModuleNotFoundError: No module named 'packaging'
" Provides: python-packaging = 23.2-3.azl3 python3-packaging = 23.2-3.azl3 python3.12-packaging = 23.2-3.azl3
```
```
$ rpm -q --provides ../out/RPMS/noarch/python3-packaging-23.2-2.azl3.noarch.rpm
python-packaging = 23.2-2.azl3
python3-packaging = 23.2-2.azl3
python3.12-packaging = 23.2-2.azl3
```

After:
```
Provides: python-packaging = 23.2-3.azl3 python3-packaging = 23.2-3.azl3 python3.12-packaging = 23.2-3.azl3 python3.12dist(packaging) = 23.2 python3dist(packaging) = 23.2
```
```
$ rpm -q --provides ../out/RPMS/noarch/python3-packaging-23.2-3.azl3.noarch.rpm
python-packaging = 23.2-3.azl3
python3-packaging = 23.2-3.azl3
python3.12-packaging = 23.2-3.azl3
python3.12dist(packaging) = 23.2
python3dist(packaging) = 23.2
```

###### Change Log  

- python-packaging: fix provides
- python-pygments: rebuild to fix provides
- python-setuptools: rebuild to fix provides 

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- https://microsoft.visualstudio.com/OS/_workitems/edit/53283659

###### Test Methodology

- Buddy build: [link](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=633509&view=results)
- Delta build: [link](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=633514&view=results)

### CODE REVIEW

File: SPECS/python-packaging/python-packaging.spec [hunk 1]
```diff
@@ -55,6 +55,9 @@ Core utilities for Python packages
 %license LICENSE
 
 %changelog
+* Thu Aug 29 2024 Andrew Phelps <anphel@microsoft.com> - 23.2-3
+- Bump release to rebuild and resolve python3dist provides issue
```
Comment (PawelWMS) [inline SPECS/python-packaging/python-packaging.spec:59]: We'd need the same bump for `python-pygments` as they are also built before `python-packaging` is installed during the toolchain build.

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh [hunk 2]
```diff
@@ -585,6 +585,9 @@ chroot_and_install_rpms pyproject-rpm-macros pyproject-srpm-macros
 # ocaml and other ocmal packages require ocaml-srpm-macros 
 build_rpm_in_chroot_no_install ocaml-srpm-macros
 
+build_rpm_in_chroot_no_install python-packaging
+chroot_and_install_rpms python-packaging python3-packaging
+# rebuild python-packaging to resolve circular dependency
 build_rpm_in_chroot_no_install python-packaging
 chroot_and_install_rpms python-packaging python3-packaging
```
Comment (PawelWMS) [inline toolkit/scripts/toolchain/build_official_toolchain_rpms.sh:592]: Nit-pick: no need to re-install as `python-packaging` doesn't need itself to build, we're just adding more `Provides` to the package but that doesn't change its functionality.

File: SPECS/python-packaging/python-packaging.spec (hunk 1 above)
Comment (PawelWMS) [inline SPECS/python-packaging/python-packaging.spec:59]: I think we should consider removing explicit `Provides` from `python-setuptools` and rebuilding it as well. This way we won't miss any provides, which may appear after we upgrade that package.

Comment (anphel31) [reply SPECS/python-packaging/python-packaging.spec:59]: Thanks, I've updated this

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh (hunk 2 above)
Comment (anphel31) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh:592]: Removed

File: SPECS/python-packaging/python-packaging.spec (hunk 1 above)
Comment (anphel31) [reply SPECS/python-packaging/python-packaging.spec:59]: I've rebuilt python-setuptools, although it still does not appear to get the automatic provides. So I've left the hard-coding for now.

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh (hunk 2 above)
Comment (PawelWMS) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh:592]: Thanks, but I think now we accidentally removed a rebuild of `python-packaging`. And we'd need a `Release` bump in the spec as well.

Comment (PawelWMS) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh:592]: The re-install is there. Did it end up making a difference and we need to keep it?

File: SPECS/python-packaging/python-packaging.spec (hunk 1 above)
Comment (PawelWMS) [reply SPECS/python-packaging/python-packaging.spec:59]: That one sounds worrying but we can investigate it separately. Could you create a bug for it, though? Fedora's spec doesn't have these particular explicit `Provides` but their RPM still lists them, so it looks like we might be missing something.

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh (hunk 2 above)
Comment (anphel31) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh:592]: Forgot to remove this part; I don't think it's necessary. Will remove and re-run the test builds

File: SPECS/python-packaging/python-packaging.spec (hunk 1 above)
Comment (anphel31) [reply SPECS/python-packaging/python-packaging.spec:59]: Sure, I've filed this tracking bug: https://microsoft.visualstudio.com/OS/_workitems/edit/53646342

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**